src/components/data-model-table/association-selection.tsx

```diff
@@ -126,7 +126,33 @@ export const AssociationSelection = <
     meta,
   };
 
-  const { tableProps, tableQuery: queryResult } = useTable<GetQuery>({
+  const searchableKeys = getSearchableKeys(associatedRecordClass);
+
+  if (searchableKeys && searchableKeys.size > 0) {
+    tableOptions['onSearch'] = (values: any) => {
+      const result = [];
+      if (!values || !values.search || values.search.length === 0) {
```

Comment:
Can we discuss how this works? I don't really get it.

Reply:
What is helpful to look at is this https://refine.dev/docs/advanced-tutorials/search/table-search/ - essentially the `onSearch` function needs to return a list of filters that refine will then use when it performs the request to graphql. The way we set it up is that these filters come from the `searchableKeys`:
```
const searchableKeys = getSearchableKeys(associatedRecordClass);
```
that we are able to extract for a given class based on the @Sortable annotations that are present on the fields on that class